### ISSUE TITLE
Should matcher be applied to detr output?

### ISSUE THREAD

**Issue by darshats** [facebookresearch/detr#510]
Hi,
I'm using DETR on custom classes (medical imaging). My question is if, after we get the model output we should use the matcher to optimize assignment of output bboxes?

I did not see that in the sample colab, but the detr eval code applies setcriterion. The reason I ask is in my output I do see very close bounding boxes emitted for two different classes (these are very close, usually 1 pixel off). These two objects in the images are alike so I could see why we might get that. The sigmoid values  of the two classes are different. Should I use the matcher here to pick the right output class?

Along the lines of line 107 of engine.py:
outputs = model(samples)
loss_dict = criterion(outputs, targets)

except that instead of computing loss, I'd have some of the code from loss_labels method of criterion to get the index permutation after matching.

**Comment by Puranjay-del-Mishra** [comment]
The model's output has loss function which uses the matcher for the loss function value. When trained well, it will predict both the bounding box's class as same, thus there's no need of manual selection like this.

**Comment by Puranjay-del-Mishra** [comment]
Can I ask you a favour. Can you check out this error/potential bug I have been facing in DETR?
link - https://github.com/facebookresearch/detr/issues/509
